float64 tran_x
float64 tran_y
string reached